### PR TITLE
Refactor Anagram

### PR DESCRIPTION
**Pull request by sozelfist**
## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I added my algorithm to the corresponding `mod.rs` file within its own folder, and in any parent folder(s).
- [x] I added my algorithm to `DIRECTORY.md` with the correct link.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

### CODE REVIEW

**Comment by sozelfist** [comment]
Can you have a look, @vil02?

File: src/string/anagram.rs [hunk 1]
```diff
@@ -1,27 +1,101 @@
...
+/// Computes the frequency of characters in a string.
+///
+/// # Arguments
+///
+/// * `s` - Input string.
+///
+/// # Returns
+///
+/// * A `HashMap` where the keys are characters and values are their frequencies.
+fn char_frequency(s: &str) -> HashMap<char, usize> {
```
Comment (vil02) [inline src/string/anagram.rs:61]:
```suggestion
fn char_count(s: &str) -> HashMap<char, usize> {
```
It is not the _frequency_.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/string/anagram.rs (hunk 1 above)
Comment (sozelfist) [reply src/string/anagram.rs:61]: Will rename the function

File: src/string/anagram.rs [hunk 2]
```diff
@@ -1,27 +1,111 @@
...
+///
+/// # Returns
+///
+/// * A `HashMap` where the keys are characters and values are their frequencies.
+fn char_count(s: &str) -> HashMap<char, usize> {
+    let mut freq = HashMap::new();
+    for c in s.chars() {
+        *freq.entry(c).or_insert(0) += 1;
+    }
+    freq
```
Comment (vil02) [inline src/string/anagram.rs:66]:
```suggestion
/// * A `HashMap` where the keys are characters and values are their count.
fn char_count(s: &str) -> HashMap<char, usize> {
    let mut res = HashMap::new();
    for c in s.chars() {
        *res.entry(c).or_insert(0) += 1;
    }
    res
```

File: src/string/anagram.rs [hunk 3]
```diff
@@ -1,27 +1,111 @@
...
+            if c.is_alphabetic() {
+                Ok(c.to_ascii_lowercase())
+            } else {
+                Err(AnagramError::NonAlphabeticCharacter)
+            }
+        })
+        .collect()
+}
+
+/// Computes the frequency of characters in a string.
```
Comment (vil02) [inline src/string/anagram.rs:52]:
```suggestion
/// Computes the histogram of characters in a string.
```

**Review by vil02 — APPROVED** [review]

**Pull request merged**